packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -851,17 +864,53 @@ class FlutterPlugin implements Plugin<Project> {
                     if (getCompileSdkFromProject(pluginProject).isInteger()) {
                         pluginCompileSdkVersion = getCompileSdkFromProject(pluginProject) as int
                     }
+
                     maxPluginCompileSdkVersion = Math.max(pluginCompileSdkVersion, maxPluginCompileSdkVersion)
+                    if (pluginCompileSdkVersion > projectCompileSdkVersion) {
+                        pluginsWithHigherSdkVersion.add(new Tuple(pluginProject.name, pluginCompileSdkVersion))
+                    }
+
                     String pluginNdkVersion = pluginProject.android.ndkVersion ?: ndkVersionIfUnspecified
                     maxPluginNdkVersion = mostRecentSemanticVersion(pluginNdkVersion, maxPluginNdkVersion)
+                    if (pluginNdkVersion != projectNdkVersion) {
+                        pluginsWithDifferentNdkVersion.add(new Tuple(pluginProject.name, pluginNdkVersion))
+                    }
 
                     numProcessedPlugins--
                     if (numProcessedPlugins == 0) {
                         if (maxPluginCompileSdkVersion > projectCompileSdkVersion) {
-                            project.logger.error("One or more plugins require a higher Android SDK version.\nFix this issue by adding the following to ${project.projectDir}${File.separator}build.gradle:\nandroid {\n  compileSdkVersion ${maxPluginCompileSdkVersion}\n  ...\n}\n")
+                            project.logger.error("Your project is configured to compile against Android SDK $projectCompileSdkVersion, but the following plugin(s) require to be compiled against a higher Android SDK version:")
+                            for (Tuple2<String, String> pluginToCompileSdkVersion : pluginsWithHigherSdkVersion) {
+                                if (pluginToCompileSdkVersion.second > projectCompileSdkVersion) {
+                                    project.logger.error("- ${pluginToCompileSdkVersion.first} compiles against Android SDK ${pluginToCompileSdkVersion.second}")
+                                }
+                            }
+                            project.logger.error("""\
+                                Fix this issue by compiling against the highest Android SDK version (they are backward compatible).
```

Comment:
Some resources I found to support the "they are backward compatible" statement:

- https://stackoverflow.com/q/31493552/7009800
- https://stackoverflow.com/q/42330768/7009800

Reply:
don't you mean to say "that are backwards compatible"

Reply:
No. I mean "Android SDK versions are backward compatible".

Reply:
okay, it wasn't so clear to me. maybe it can be clearer

Reply:
> - https://stackoverflow.com/q/42330768/7009800

I think it'd be great to include the guide linked here in the error message.

Maybe something like this could work:

```
Almost all changes to the Android framework API are additive, so older plugins
can generally be compiled with the latest SDK version without compatibility issues.
(see https://developer.android.com/guide/topics/manifest/uses-sdk-element#fc)

Consider using the latest SDK version listed, or changing/removing the above dependencies.
```

Reply:
I'm not sure if the wording is right here:

> so older plugins can generally be compiled with the latest SDK version without compatibility issues.

The problem in this case is not "older plugins", but the app that compiles using Android SDK that is too low.